integrations/tensorflow/e2e/python_outputs_test.py

```diff
@@ -0,0 +1,133 @@
+# Lint as: python3
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests of tf.functions returning common python outputs types."""
+
+from absl import app
+import numpy as np
+from pyiree.tf.support import tf_test_utils
+import tensorflow.compat.v2 as tf
+
+
+class PythonOutputsModule(tf.Module):
+
+  @tf.function(input_signature=[tf.TensorSpec([2], tf.float32)])
+  def tensor_output(self, a):
+    return a
+
+  @tf.function(input_signature=[tf.TensorSpec([2], tf.float32)])
+  def tuple_output_len_one(self, a):
+    return (a,)
+
+  @tf.function(input_signature=[tf.TensorSpec([2], tf.float32)])
+  def tuple_output_len_two(self, a):
+    return (a, a * a)
+
+  @tf.function(input_signature=[tf.TensorSpec([2], tf.float32)])
+  def tuple_output_len_three(self, a):
+    return (a, a * a, a * a * a)
```

Comment:
nit: instead of using `a*a` and `a*a*a` as the other ones, using `2*a` and `3*a` will generally produce IR that is easier to follow.

I think you can quickly find/replace it:
s/a * a * a/3 *a/
s/a * a/2 * a/